airflow/providers/cncf/kubernetes/operators/pod.py

```diff
@@ -822,8 +823,12 @@ def process_pod_deletion(self, pod: k8s.V1Pod, *, reraise=True):
                     self.log.info("Deleting pod: %s", pod.metadata.name)
                     self.pod_manager.delete_pod(pod)
                 elif should_delete_pod and istio_enabled:
-                    self.log.info("Deleting istio-proxy sidecar inside %s: ", pod.metadata.name)
-                    self.kill_istio_sidecar(pod)
+                    if container_is_running(pod, self.ISTIO_CONTAINER_NAME):
+                        self.log.info("Deleting istio-proxy sidecar inside %s: ", pod.metadata.name)
+                        self.kill_istio_sidecar(pod)
+                        self.pod_manager.delete_pod(pod)
+                    else:
+                        self.pod_manager.delete_pod(pod)
```

Comment:
```suggestion
                        self.pod_manager.delete_pod(pod)
```

Doesn't the sidecar listen to sigterm from k8s? In fact this whole thing can be simplified further, if so (no need to see if an istio sidecar is running).

Reply:
hmmm yes. In my istio version I can see that `delete_namespaced_pod` works even when sidecar is running, and seems to me that we don't actually need to `kill_istio_sidecar` at all. However I'm not sure if this behaviour also applies to older version of istio. Shall we keep the logic to end istio sidecar followed by deleting the pod ? Or we should just go ahead to delete the pod directly

Reply:
Indeed, when we attempt to terminate the pod, all the containers receive a SIGTERM signal. They manage this through a signal handler. Kubernetes will wait a maximum of `graceperiod` seconds before resorting to a SIGKILL to forcefully terminate them. Given that the KPO has the `termination_grace_period` argument, I concur with Jed's suggestion with a small change; no need for `if should_delete_pod and not istio_enabled` and `if should_delete_pod and istio_enabled`, all this block could be deleted, and we can kill the pod when `should_delete_pod` is Truee regardless the value of `istio_enabled`.

Reply:
Thanks both for your feedback. I've adopted the suggestion and proceed for deletion whenever `should_delete_pod` is true.